src/main/java/com/teamwizardry/refraction/common/effect/EffectBurn.java

```diff
@@ -68,13 +73,17 @@ public void runEntity(World world, Entity entity, int potency) {
 		if (entity instanceof EntityItem) {
 			EntityItem item = (EntityItem) entity;
 			ItemStack stack = item.getItem();
-			if (FurnaceRecipes.instance().getSmeltingResult(stack) != ItemStack.EMPTY) {
-				if (ThreadLocalRandom.current().nextInt(100) == 0) {
-					ItemStack result = FurnaceRecipes.instance().getSmeltingResult(stack);
-					EntityItem cooked = new EntityItem(world, item.posX, item.posY, item.posZ);
-					cooked.dropItem(result.getItem(), 1);
+			if (!world.isRemote) {
+				ItemStack result = FurnaceRecipes.instance().getSmeltingResult(stack);
+				if (!result.isEmpty()) {
+					EntityItem cooked = new EntityItem(world, item.posX, item.posY, item.posZ, result.copy());
 					cooked.setNoPickupDelay();
-					stack.setCount(stack.getCount()-1);
+					cooked.motionX = entity.motionX;
+					cooked.motionY = entity.motionY;
+					cooked.motionZ = entity.motionZ;
+		            world.spawnEntity(cooked);
+					stack.shrink(1);
+					item.setItem(stack);
```

Comment:
Are you sure this line is necessary? it's still the original stack you are working with.

If it truely is not necessary, there is also no need anymore to have the "item" variable.

Reply:
Yes it will sync changes to client.

Reply:
sadly you are right, and i don't see a better way to do it.